portability-api-launcher/src/main/java/org/dataportabilityproject/api/launcher/ExtensionContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2018 The Data-Portability Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataportabilityproject.api.launcher;
+
+/**
+ * Provides information required to bootstrap extensions.
+ */
+public interface ExtensionContext {
+
+    /**
+     * Returns the system logger extension service can use to record events.
+     */
+    Logger getLogger();
+
+    /**
+     * Returns a system service such as a type mapper extension services may require.
```

Comment:
Just for clarity, what would be a specific example?

Reply:
Nothing requires Java 9 so it should be good.

Do you mean an example of the logger or ExtensionContext or both? Let me know and I'll create a Gist.

Reply:
Oh, I mean what would be an example of a system service?

Reply:
A system service could be a registry or other service that extensions must register against. For example, an extension may need to register some type of provider.